### DISCUSSION TITLE
[Bug]: Question: how to use multiple gpu for inferencing

### DISCUSSION THREAD

**Discussion by treya-lin** [speechbrain/speechbrain#1736] [Q&A]
### Describe the bug

Hi, I loaded the model with codes below:
```python
os.environ["CUDA_VISIBLE_DEVICES"] = '0,1,2,3'
verification = SpeakerRecognition.from_hparams(
    source="speechbrain/spkrec-ecapa-voxceleb", \
    savedir="pretrained_models/spkrec-ecapa-voxceleb", \
    run_opts={"device":"cuda","data_parallel_count":4,"data_parallel_backend":True}
)
```
I have 4 gpus, and I want them all to be used to speed up inferencing. I assume setting data_parallel_count as -1 or 4 will result in the model replicated on all gpus, is that correct? But for now, after I run the code, it appears on one gpu only:

```
|===============================+======================+======================|
|   0  GeForce GTX 108...  Off  | 00000000:02:00.0 Off |                  N/A |
| 23%   24C    P2    53W / 250W |    963MiB / 11177MiB |      0%   E. Process |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   1  GeForce GTX 108...  Off  | 00000000:03:00.0 Off |                  N/A |
| 23%   24C    P8     8W / 250W |      2MiB / 11178MiB |      0%   E. Process |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   2  GeForce GTX 108...  Off  | 00000000:82:00.0 Off |                  N/A |
| 23%   28C    P8     8W / 250W |      2MiB / 11178MiB |      0%   E. Process |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   3  GeForce GTX 108...  Off  | 00000000:83:00.0 Off |                  N/A |
| 23%   25C    P8     8W / 250W |      2MiB / 11178MiB |      0%   E. Process |
```

May I know how to make the multi-gpu inferencing work? (It has been working properly on single gpu.)
Thanks!

### Expected behaviour

I need to extract speaker embeddings for a large amount of wav files, so I hope the inferencing can be done on multiple gpus.

**Comment by asumagic** [comment]
I believe DP support is [being phased away by both PyTorch and SpeechBrain](https://github.com/speechbrain/speechbrain/issues/1713#issuecomment-1324344860) and is generally recommended against.

[DDP is the recommended option](https://speechbrain.readthedocs.io/en/latest/multigpu.html#multi-gpu-training-using-distributed-data-parallel-ddp)... for training. I'm not sure how it can be applied to inference, if it is possible at all in SB right now.

@Adel-Moumen I think you've used DDP a fair bit, thoughts?

**Comment by TParcollet** [comment]
@Gastron do you know if our pretrained interface supports DDP right now?
